src/components/LearnerActivityTable/data/hooks/useCourseEnrollments.js

```diff
@@ -0,0 +1,88 @@
+import {
+  useCallback, useMemo, useRef, useState,
+} from 'react';
+import { camelCaseObject } from '@edx/frontend-platform/utils';
+import debounce from 'lodash.debounce';
+import { logError } from '@edx/frontend-platform/logging';
+import EnterpriseDataApiService from '../../../../data/services/EnterpriseDataApiService';
+import EVENT_NAMES from '../../../../eventTracking';
+import { trackDataTableEvent } from '../utils';
+
+const applySortByToOptions = (sortBy, apiFieldsForColumnAccessor, options) => {
```

Comment:
🥳  Nice job bringing this in.  Could this be the beginnings of a `useDatatable` hook with this simple function?

Reply:
Thanks.
Yes, I initially implemented the necessary functionalities, and this can be updated over time as more requirements arise.